backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/ElasticsearchDistanceToFieldProjection.java

```diff
@@ -29,18 +31,36 @@
 
 	private static final Pattern NON_DIGITS_PATTERN = Pattern.compile( "\\D" );
 
+	private static final String DISTANCE_PROJECTION_SCRIPT =
+		// Use ".size() != 0" to check whether this field has a value. ".value != null" won't work on ES7+
+		" Object result;" +
+		" if (doc[params.fieldPath].size() != 0) {" +
+			" result = doc[params.fieldPath].arcDistance(params.lat, params.lon);" +
+		" } else {" +
+			" for (int i=0; i<params.nestedPaths.length; i++) {" +
+				" String nestedPath = params.nestedPaths[i];" +
+				" String relativeFieldPath = params.relativeFieldPaths[i];" +
+				" if (params['_source'][nestedPath] == null) continue;" +
+				" if (params['_source'][nestedPath][relativeFieldPath] == null) continue;" +
+				" return params['_source'][nestedPath][relativeFieldPath]" +
```

Comment:
Maybe add a comment here to explain why we have to go to such lengths?

Reply:
I will